SnaffCore/ActiveDirectory/AdData.cs

```diff
@@ -60,40 +78,38 @@ public AdData()
             {
                 Mq.Trace("Target DC and Domain specified: " + MyOptions.TargetDomain + " + " + MyOptions.TargetDc);
                 _targetDc = MyOptions.TargetDc;
-                _targetDomain = MyOptions.TargetDomain;
+                _targetDomain = Domain.GetDomain(new DirectoryContext(DirectoryContextType.Domain, MyOptions.TargetDomain));
             }
             // no target DC or domain set
             else
             {
                 Mq.Trace("Getting current domain from user context.");
-                _currentDomain = Domain.GetCurrentDomain();
-                _targetDomain = _currentDomain.Name;
-                _targetDc = _targetDomain;
+//                _currentDomain = Domain.GetCurrentDomain();
+//                _targetDomain = _currentDomain.Name;
+                _targetDomain = Domain.GetCurrentDomain();
+                _targetDc = _targetDomain.PdcRoleOwner.Name;
```

Comment:
This is 'more accurate' but I'm pretty sure it'll actually be quite a bit slower than if the PDC is e.g. in a different geographical region than the user, with a slow link between the two - that's why i was just giving it the domain name and letting windows figure it out.